src/pointer-analysis/value_set.cpp

```diff
@@ -928,17 +923,16 @@ void value_sett::get_value_set_rec(
   else if(expr.id()==ID_byte_extract_little_endian ||
           expr.id()==ID_byte_extract_big_endian)
   {
-    if(expr.operands().size()!=2)
-      throw "byte_extract takes two operands";
+    const auto &byte_extract_expr = to_byte_extract_expr(expr);
 
     bool found=false;
 
-    exprt op1=expr.op1();
+    exprt op1 = byte_extract_expr.op1();
     if(eval_pointer_offset(op1, ns))
       simplify(op1, ns);
 
     const auto op1_offset = numeric_cast<mp_integer>(op1);
-    const typet &op0_type=ns.follow(expr.op0().type());
+    const typet &op0_type = ns.follow(byte_extract_expr.op0().type());
```

Comment:
Use `byte_extract_expr.op()`

Reply:
This hasn't been done. And I see now that the variable should be renamed to `op_type`.